<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />

<meta name="viewport" content="width=device-width, initial-scale=1">

<meta name="author" content="Jianfeng Li" />

<meta name="date" content="2018-01-04" />

<title>Introduction to ngstk</title>



<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
table.sourceCode, tr.sourceCode, td.lineNumbers, td.sourceCode {
  margin: 0; padding: 0; vertical-align: baseline; border: none; }
table.sourceCode { width: 100%; line-height: 100%; }
td.lineNumbers { text-align: right; padding-right: 4px; padding-left: 4px; color: #aaaaaa; border-right: 1px solid #aaaaaa; }
td.sourceCode { padding-left: 5px; }
code > span.kw { color: #007020; font-weight: bold; }
code > span.dt { color: #902000; }
code > span.dv { color: #40a070; }
code > span.bn { color: #40a070; }
code > span.fl { color: #40a070; }
code > span.ch { color: #4070a0; }
code > span.st { color: #4070a0; }
code > span.co { color: #60a0b0; font-style: italic; }
code > span.ot { color: #007020; }
code > span.al { color: #ff0000; font-weight: bold; }
code > span.fu { color: #06287e; }
code > span.er { color: #ff0000; font-weight: bold; }
</style>



<link href="data:text/css,body%20%7B%0A%20%20background%2Dcolor%3A%20%23fff%3B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20max%2Dwidth%3A%20700px%3B%0A%20%20overflow%3A%20visible%3B%0A%20%20padding%2Dleft%3A%202em%3B%0A%20%20padding%2Dright%3A%202em%3B%0A%20%20font%2Dfamily%3A%20%22Open%20Sans%22%2C%20%22Helvetica%20Neue%22%2C%20Helvetica%2C%20Arial%2C%20sans%2Dserif%3B%0A%20%20font%2Dsize%3A%2014px%3B%0A%20%20line%2Dheight%3A%201%2E35%3B%0A%7D%0A%0A%23header%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0A%0A%23TOC%20%7B%0A%20%20clear%3A%20both%3B%0A%20%20margin%3A%200%200%2010px%2010px%3B%0A%20%20padding%3A%204px%3B%0A%20%20width%3A%20400px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20border%2Dradius%3A%205px%3B%0A%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20font%2Dsize%3A%2013px%3B%0A%20%20line%2Dheight%3A%201%2E3%3B%0A%7D%0A%20%20%23TOC%20%2Etoctitle%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%20%20font%2Dsize%3A%2015px%3B%0A%20%20%20%20margin%2Dleft%3A%205px%3B%0A%20%20%7D%0A%0A%20%20%23TOC%20ul%20%7B%0A%20%20%20%20padding%2Dleft%3A%2040px%3B%0A%20%20%20%20margin%2Dleft%3A%20%2D1%2E5em%3B%0A%20%20%20%20margin%2Dtop%3A%205px%3B%0A%20%20%20%20margin%2Dbottom%3A%205px%3B%0A%20%20%7D%0A%20%20%23TOC%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dleft%3A%20%2D2em%3B%0A%20%20%7D%0A%20%20%23TOC%20li%20%7B%0A%20%20%20%20line%2Dheight%3A%2016px%3B%0A%20%20%7D%0A%0Atable%20%7B%0A%20%20margin%3A%201em%20auto%3B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dcolor%3A%20%23DDDDDD%3B%0A%20%20border%2Dstyle%3A%20outset%3B%0A%20%20border%2Dcollapse%3A%20collapse%3B%0A%7D%0Atable%20th%20%7B%0A%20%20border%2Dwidth%3A%202px%3B%0A%20%20padding%3A%205px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%7D%0Atable%20td%20%7B%0A%20%20border%2Dwidth%3A%201px%3B%0A%20%20border%2Dstyle%3A%20inset%3B%0A%20%20line%2Dheight%3A%2018px%3B%0A%20%20padding%3A%205px%205px%3B%0A%7D%0Atable%2C%20table%20th%2C%20table%20td%20%7B%0A%20%20border%2Dleft%2Dstyle%3A%20none%3B%0A%20%20border%2Dright%2Dstyle%3A%20none%3B%0A%7D%0Atable%20thead%2C%20table%20tr%2Eeven%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Ap%20%7B%0A%20%20margin%3A%200%2E5em%200%3B%0A%7D%0A%0Ablockquote%20%7B%0A%20%20background%2Dcolor%3A%20%23f6f6f6%3B%0A%20%20padding%3A%200%2E25em%200%2E75em%3B%0A%7D%0A%0Ahr%20%7B%0A%20%20border%2Dstyle%3A%20solid%3B%0A%20%20border%3A%20none%3B%0A%20%20border%2Dtop%3A%201px%20solid%20%23777%3B%0A%20%20margin%3A%2028px%200%3B%0A%7D%0A%0Adl%20%7B%0A%20%20margin%2Dleft%3A%200%3B%0A%7D%0A%20%20dl%20dd%20%7B%0A%20%20%20%20margin%2Dbottom%3A%2013px%3B%0A%20%20%20%20margin%2Dleft%3A%2013px%3B%0A%20%20%7D%0A%20%20dl%20dt%20%7B%0A%20%20%20%20font%2Dweight%3A%20bold%3B%0A%20%20%7D%0A%0Aul%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%7D%0A%20%20ul%20li%20%7B%0A%20%20%20%20list%2Dstyle%3A%20circle%20outside%3B%0A%20%20%7D%0A%20%20ul%20ul%20%7B%0A%20%20%20%20margin%2Dbottom%3A%200%3B%0A%20%20%7D%0A%0Apre%2C%20code%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20color%3A%20%23333%3B%0A%20%20white%2Dspace%3A%20pre%2Dwrap%3B%20%20%20%20%2F%2A%20Wrap%20long%20lines%20%2A%2F%0A%7D%0Apre%20%7B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20margin%3A%205px%200px%2010px%200px%3B%0A%20%20padding%3A%2010px%3B%0A%7D%0Apre%3Anot%28%5Bclass%5D%29%20%7B%0A%20%20background%2Dcolor%3A%20%23f7f7f7%3B%0A%7D%0A%0Acode%20%7B%0A%20%20font%2Dfamily%3A%20Consolas%2C%20Monaco%2C%20%27Courier%20New%27%2C%20monospace%3B%0A%20%20font%2Dsize%3A%2085%25%3B%0A%7D%0Ap%20%3E%20code%2C%20li%20%3E%20code%20%7B%0A%20%20padding%3A%202px%200px%3B%0A%7D%0A%0Adiv%2Efigure%20%7B%0A%20%20text%2Dalign%3A%20center%3B%0A%7D%0Aimg%20%7B%0A%20%20background%2Dcolor%3A%20%23FFFFFF%3B%0A%20%20padding%3A%202px%3B%0A%20%20border%3A%201px%20solid%20%23DDDDDD%3B%0A%20%20border%2Dradius%3A%203px%3B%0A%20%20border%3A%201px%20solid%20%23CCCCCC%3B%0A%20%20margin%3A%200%205px%3B%0A%7D%0A%0Ah1%20%7B%0A%20%20margin%2Dtop%3A%200%3B%0A%20%20font%2Dsize%3A%2035px%3B%0A%20%20line%2Dheight%3A%2040px%3B%0A%7D%0A%0Ah2%20%7B%0A%20%20border%2Dbottom%3A%204px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20padding%2Dbottom%3A%202px%3B%0A%20%20font%2Dsize%3A%20145%25%3B%0A%7D%0A%0Ah3%20%7B%0A%20%20border%2Dbottom%3A%202px%20solid%20%23f7f7f7%3B%0A%20%20padding%2Dtop%3A%2010px%3B%0A%20%20font%2Dsize%3A%20120%25%3B%0A%7D%0A%0Ah4%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23f7f7f7%3B%0A%20%20margin%2Dleft%3A%208px%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Ah5%2C%20h6%20%7B%0A%20%20border%2Dbottom%3A%201px%20solid%20%23ccc%3B%0A%20%20font%2Dsize%3A%20105%25%3B%0A%7D%0A%0Aa%20%7B%0A%20%20color%3A%20%230033dd%3B%0A%20%20text%2Ddecoration%3A%20none%3B%0A%7D%0A%20%20a%3Ahover%20%7B%0A%20%20%20%20color%3A%20%236666ff%3B%20%7D%0A%20%20a%3Avisited%20%7B%0A%20%20%20%20color%3A%20%23800080%3B%20%7D%0A%20%20a%3Avisited%3Ahover%20%7B%0A%20%20%20%20color%3A%20%23BB00BB%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22http%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%20%20a%5Bhref%5E%3D%22https%3A%22%5D%20%7B%0A%20%20%20%20text%2Ddecoration%3A%20underline%3B%20%7D%0A%0A%2F%2A%20Class%20described%20in%20https%3A%2F%2Fbenjeffrey%2Ecom%2Fposts%2Fpandoc%2Dsyntax%2Dhighlighting%2Dcss%0A%20%20%20Colours%20from%20https%3A%2F%2Fgist%2Egithub%2Ecom%2Frobsimmons%2F1172277%20%2A%2F%0A%0Acode%20%3E%20span%2Ekw%20%7B%20color%3A%20%23555%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Keyword%20%2A%2F%0Acode%20%3E%20span%2Edt%20%7B%20color%3A%20%23902000%3B%20%7D%20%2F%2A%20DataType%20%2A%2F%0Acode%20%3E%20span%2Edv%20%7B%20color%3A%20%2340a070%3B%20%7D%20%2F%2A%20DecVal%20%28decimal%20values%29%20%2A%2F%0Acode%20%3E%20span%2Ebn%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20BaseN%20%2A%2F%0Acode%20%3E%20span%2Efl%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Float%20%2A%2F%0Acode%20%3E%20span%2Ech%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20Char%20%2A%2F%0Acode%20%3E%20span%2Est%20%7B%20color%3A%20%23d14%3B%20%7D%20%2F%2A%20String%20%2A%2F%0Acode%20%3E%20span%2Eco%20%7B%20color%3A%20%23888888%3B%20font%2Dstyle%3A%20italic%3B%20%7D%20%2F%2A%20Comment%20%2A%2F%0Acode%20%3E%20span%2Eot%20%7B%20color%3A%20%23007020%3B%20%7D%20%2F%2A%20OtherToken%20%2A%2F%0Acode%20%3E%20span%2Eal%20%7B%20color%3A%20%23ff0000%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20AlertToken%20%2A%2F%0Acode%20%3E%20span%2Efu%20%7B%20color%3A%20%23900%3B%20font%2Dweight%3A%20bold%3B%20%7D%20%2F%2A%20Function%20calls%20%2A%2F%20%0Acode%20%3E%20span%2Eer%20%7B%20color%3A%20%23a61717%3B%20background%2Dcolor%3A%20%23e3d2d2%3B%20%7D%20%2F%2A%20ErrorTok%20%2A%2F%0A%0A" rel="stylesheet" type="text/css" />

</head>

<body>




<h1 class="title toc-ignore">Introduction to ngstk</h1>
<h4 class="author"><em>Jianfeng Li</em></h4>
<h4 class="date"><em>2018-01-04</em></h4>



<div id="introduction" class="section level2">
<h2>Introduction</h2>
<p><a href="https://github.com/JhuangLab/ngstk">ngstk</a> is an R package to facilitate the analysis of NGS data, such as visualization, conversion of the data format for WEB service input and another purpose.</p>
<p>You can learn some usage of ngstk through this tutorial.</p>
</div>
<div id="usage" class="section level2">
<h2>Usage</h2>
<div id="data-format-conversion" class="section level3">
<h3>Data format conversion</h3>
<p>We defined a rule to facilitate the data format conversion. It is a good choice to save the all of output colums meta information. Moreover, the other input data should establish a connection with the output colums according the requirement.</p>
<ul>
<li>alias, connect the two table by colnames</li>
<li>extract_pattern, extract defined pattern from one of colum</li>
<li>raw and new, repace the raw value with the new value</li>
<li>na.replace, replace the NA value to defined value</li>
</ul>
<p>The follow example configuration file can be used to convert <a href="https://github.com/JhuangLab/iseq">iseq</a>, a pipeline to analysis genetic variants from NGS data, output data to the web service <a href="https://proteinpaint.stjude.org/">ProteinPaint</a>, a tool to visulize the mutation data.</p>
<pre class="toml"><code>Title = &quot;Proteinpaint configuration file&quot;

# muts2pp is a function that can convert mutation data to Proteinpaint input format
[muts2pp.meta.defined_cols]
colnames = [&quot;gene&quot;, &quot;refseq&quot;, &quot;chromosome&quot;, &quot;start&quot;, &quot;aachange&quot;, &quot;class&quot;, &quot;disease&quot;, &quot;sample&quot;]
handler_lib = &quot;default_handlers&quot;
mhandler_lib = &quot;default_mhandlers&quot;
[muts2pp.meta.defined_cols.description]
gene = &quot;Gene symbol, e.g TP53, PTEN&quot;
refseq = &quot;Transcript of refSeq or Ensemble, e.g NM_000546, ENST00000635293&quot;
chromosome = &quot;Chromosome, e.g. chr1, chr2&quot;
start = &quot;Chromosome start location of a mutation site, e.g. 153249385&quot;
aachange = &quot;Amino acid level change of gene mutation, e.g. p.R347C, p.L615delinsDL&quot;
class = &quot;Mutation type, e.g. nonsense, proteinIns, proteinDel, frameshift&quot;
disease = &quot;Disease name or sample group name, e.g. B-ALL, T-ALL, G1, G2&quot;
sample = &quot;Sample name&quot;

[muts2pp.format.iseq.gene]
alias = [&quot;gene&quot;, &quot;symbol&quot;]

[muts2pp.format.iseq.refseq]
alias = [&quot;refseq&quot;, &quot;transcription.id&quot;]

[muts2pp.format.iseq.chromosome]
alias = [&quot;chromosome&quot;]

[muts2pp.format.iseq.start]
alias = [&quot;start&quot;]

[muts2pp.format.iseq.aachange]
alias = [&quot;aachange&quot;, &quot;amino.acid.change&quot;]
extract_pattern = &quot;p[.]+.*$&quot;

[muts2pp.format.iseq.class]
alias = [&quot;mutation_type&quot;, &quot;mutation.type&quot;]
raw = [&quot;nonframeshift ins&quot;, &quot;nonframeshift del&quot;,
       &quot;frameshift ins&quot;, &quot;frameshift del&quot;, &quot;stoploss&quot;, &quot;nonsense&quot;,
             &quot;splice&quot;]
new = [&quot;proteinIns&quot;, &quot;proteinDel&quot;, &quot;frameshift&quot;, &quot;frameshift&quot;,
       &quot;nonsense&quot;, &quot;nonsense&quot;, &quot;splice&quot;]
na_replace = &quot;splice&quot;</code></pre>
<p>muts2pp, muts2mutation_mapper, muts2oncoprinter and fusions2pp are the example functions based on the established rules.</p>
<pre class="sourceCode r"><code class="sourceCode r">demo_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;demo/proteinpaint/muts2pp_iseq.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;ngstk&quot;</span>)
input_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(demo_file, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
disease &lt;-<span class="st"> &quot;T-ALL&quot;</span>
input_data &lt;-<span class="st"> </span><span class="kw">data.frame</span>(input_data, disease)
input_data$disease &lt;-<span class="st"> </span><span class="kw">as.character</span>(input_data$disease)

<span class="co"># Convert mutations data to proteinpaint input</span>
result &lt;-<span class="st"> </span><span class="kw">muts2pp</span>(input_data, <span class="dt">input_type =</span> <span class="st">&quot;iseq&quot;</span>)
<span class="kw">head</span>(result)
<span class="co">#&gt;    gene       refseq chromosome     start aachange      class disease</span>
<span class="co">#&gt; 1  IDH1 NM_001282386       chr2 209113113  p.R132S   missense   T-ALL</span>
<span class="co">#&gt; 2 PTPRC    NM_080921       chr1 198711362  p.A694T   missense   T-ALL</span>
<span class="co">#&gt; 3 PTPRC    NM_080921       chr1 198685843 p.I281fs frameshift   T-ALL</span>
<span class="co">#&gt; 4  ASPM NM_001206846       chr1 197093449 p.L1061F   missense   T-ALL</span>
<span class="co">#&gt; 5  ASPM    NM_018136       chr1 197070987 p.Q2465R   missense   T-ALL</span>
<span class="co">#&gt; 6  FAT1    NM_005245       chr4 187630354  p.L210F   missense   T-ALL</span>
<span class="co">#&gt;   sample</span>
<span class="co">#&gt; 1     A1</span>
<span class="co">#&gt; 2     A2</span>
<span class="co">#&gt; 3     A3</span>
<span class="co">#&gt; 4     A4</span>
<span class="co">#&gt; 5     A5</span>
<span class="co">#&gt; 6     A6</span>
<span class="co"># Convert mutations data to cbioportal input</span>
result &lt;-<span class="st"> </span><span class="kw">muts2mutation_mapper</span>(input_data, <span class="dt">input_type =</span> <span class="st">&quot;iseq&quot;</span>)
<span class="co">#&gt; Warning in handler(handler_data, config_input, defined_cols, input_data, :</span>
<span class="co">#&gt; Validation_Status were not exists or not be recognize correctly in input</span>
<span class="co">#&gt; data!</span>
<span class="co">#&gt; Warning in handler(handler_data, config_input, defined_cols, input_data, :</span>
<span class="co">#&gt; Mutation_Status were not exists or not be recognize correctly in input</span>
<span class="co">#&gt; data!</span>
<span class="co">#&gt; Warning in handler(handler_data, config_input, defined_cols, input_data, :</span>
<span class="co">#&gt; Center were not exists or not be recognize correctly in input data!</span>
<span class="kw">head</span>(result)
<span class="co">#&gt;   Hugo_Symbol Sample_ID Protein_Change     Mutation_Type Chromosome</span>
<span class="co">#&gt; 1        IDH1        A1          R132S Missense_Mutation       chr2</span>
<span class="co">#&gt; 2       PTPRC        A2          A694T Missense_Mutation       chr1</span>
<span class="co">#&gt; 3       PTPRC        A3         I281fs   Frame_Shift_Ins       chr1</span>
<span class="co">#&gt; 4        ASPM        A4         L1061F Missense_Mutation       chr1</span>
<span class="co">#&gt; 5        ASPM        A5         Q2465R Missense_Mutation       chr1</span>
<span class="co">#&gt; 6        FAT1        A6          L210F Missense_Mutation       chr4</span>
<span class="co">#&gt;   Start_Position End_Position Reference_Allele Variant_Allele</span>
<span class="co">#&gt; 1      209113113    209113113                G              T</span>
<span class="co">#&gt; 2      198711362    198711362                G              A</span>
<span class="co">#&gt; 3      198685843    198685843                -              T</span>
<span class="co">#&gt; 4      197093449    197093449                G              A</span>
<span class="co">#&gt; 5      197070987    197070987                T              C</span>
<span class="co">#&gt; 6      187630354    187630354                G              A</span>
<span class="co">#&gt;   Validation_Status Mutation_Status Center</span>
<span class="co">#&gt; 1                NA              NA     NA</span>
<span class="co">#&gt; 2                NA              NA     NA</span>
<span class="co">#&gt; 3                NA              NA     NA</span>
<span class="co">#&gt; 4                NA              NA     NA</span>
<span class="co">#&gt; 5                NA              NA     NA</span>
<span class="co">#&gt; 6                NA              NA     NA</span>
result &lt;-<span class="st"> </span><span class="kw">muts2oncoprinter</span>(input_data, <span class="dt">input_type =</span> <span class="st">&quot;iseq&quot;</span>)
<span class="kw">head</span>(result)
<span class="co">#&gt;   Sample  Gene Alteration     Type</span>
<span class="co">#&gt; 1     A1  IDH1      R132S MISSENSE</span>
<span class="co">#&gt; 2     A2 PTPRC      A694T MISSENSE</span>
<span class="co">#&gt; 3     A3 PTPRC     I281fs    TRUNC</span>
<span class="co">#&gt; 4     A4  ASPM     L1061F MISSENSE</span>
<span class="co">#&gt; 5     A5  ASPM     Q2465R MISSENSE</span>
<span class="co">#&gt; 6     A6  FAT1      L210F MISSENSE</span>

demo_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="st">'demo/proteinpaint/fusions2pp_fusioncatcher.txt'</span>, <span class="dt">package =</span> <span class="st">'ngstk'</span>)
input_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(demo_file, <span class="dt">sep =</span> <span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
disease &lt;-<span class="st"> 'B-ALL'</span>
sampletype &lt;-<span class="st"> 'diagnose'</span>
input_data &lt;-<span class="st"> </span><span class="kw">data.frame</span>(input_data, disease, sampletype)
input_data$disease &lt;-<span class="st"> </span><span class="kw">as.character</span>(input_data$disease)
<span class="co"># Convert fusions data to proteinpaint input</span>
result &lt;-<span class="st"> </span><span class="kw">fusions2pp</span>(input_data, <span class="dt">input_type =</span> <span class="st">'fusioncatcher'</span>)
<span class="kw">head</span>(result)
<span class="co">#&gt;   disease sampletype gene_a        refseq_a chr_a position_a gene_b</span>
<span class="co">#&gt; 1   B-ALL   diagnose   TCF3 ENSG00000071564 chr19    1619110   PBX1</span>
<span class="co">#&gt; 2   B-ALL   diagnose   TCF3 ENSG00000071564 chr19    1619110   PBX1</span>
<span class="co">#&gt; 3   B-ALL   diagnose   TCF3 ENSG00000071564 chr19    1619110   PBX1</span>
<span class="co">#&gt; 4   B-ALL   diagnose   TCF3 ENSG00000071564 chr19    1619110   PBX1</span>
<span class="co">#&gt; 5   B-ALL   diagnose   TCF3 ENSG00000071564 chr19    1619110   PBX1</span>
<span class="co">#&gt; 6   B-ALL   diagnose   GYPE ENSG00000197465  chr4  144801564   GYPA</span>
<span class="co">#&gt;          refseq_b chr_b position_b strand_a strand_b patient</span>
<span class="co">#&gt; 1 ENSG00000185630  chr1  164761731        -        +      A1</span>
<span class="co">#&gt; 2 ENSG00000185630  chr1  164761731        -        +      A2</span>
<span class="co">#&gt; 3 ENSG00000185630  chr1  164761731        -        +      A3</span>
<span class="co">#&gt; 4 ENSG00000185630  chr1  164704095        -        +      A4</span>
<span class="co">#&gt; 5 ENSG00000185630  chr1  164704095        -        +      A5</span>
<span class="co">#&gt; 6 ENSG00000170180  chr4  145040934        -        -      A6</span></code></pre>
<p>merge_table_files is the another util function to merge multiple table files.</p>
<pre class="sourceCode r"><code class="sourceCode r">a &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">col1=</span><span class="dv">1</span>:<span class="dv">6</span>, <span class="dt">col2=</span><span class="dv">2</span>:<span class="dv">7</span>)
b &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">col1=</span><span class="dv">6</span>:<span class="dv">11</span>, <span class="dt">col2=</span><span class="dv">1</span>:<span class="dv">6</span>)
file_a &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">tempfile</span>(), <span class="st">'_abcd'</span>)
file_b &lt;-<span class="st"> </span><span class="kw">paste0</span>(<span class="kw">tempfile</span>(), <span class="st">'_abcd'</span>)
<span class="kw">write.table</span>(a, file_a, <span class="dt">sep =</span> <span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)
<span class="kw">write.table</span>(b, file_b, <span class="dt">sep =</span> <span class="st">'</span><span class="ch">\t</span><span class="st">'</span>, <span class="dt">row.names =</span> <span class="ot">FALSE</span>)
input_files &lt;-<span class="st"> </span><span class="kw">c</span>(file_a, file_b)
x1 &lt;-<span class="st"> </span><span class="kw">merge_table_files</span>(<span class="dt">input_files =</span> input_files)
<span class="kw">head</span>(x1)
<span class="co">#&gt;                                 filename col1 col2</span>
<span class="co">#&gt; 1 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    1    2</span>
<span class="co">#&gt; 2 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    2    3</span>
<span class="co">#&gt; 3 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    3    4</span>
<span class="co">#&gt; 4 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    4    5</span>
<span class="co">#&gt; 5 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    5    6</span>
<span class="co">#&gt; 6 /tmp/RtmpUrg5Db/file18ba97fafd4cf_abcd    6    7</span>
x2 &lt;-<span class="st"> </span><span class="kw">merge_table_files</span>(<span class="dt">files_dir =</span> <span class="kw">tempdir</span>(), <span class="dt">pattern =</span> <span class="st">'.*_abcd$'</span>)
<span class="kw">head</span>(x2)
<span class="co">#&gt;                                filename col1 col2</span>
<span class="co">#&gt; 1 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    6    1</span>
<span class="co">#&gt; 2 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    7    2</span>
<span class="co">#&gt; 3 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    8    3</span>
<span class="co">#&gt; 4 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    9    4</span>
<span class="co">#&gt; 5 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd   10    5</span>
<span class="co">#&gt; 6 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd   11    6</span>
outfn =<span class="st"> </span><span class="kw">tempfile</span>()
x3 &lt;-<span class="st"> </span><span class="kw">merge_table_files</span>(<span class="dt">files_dir =</span> <span class="kw">tempdir</span>(), <span class="dt">pattern =</span> <span class="st">&quot;.*_abcd$&quot;</span>, <span class="dt">outfn =</span> outfn)
<span class="kw">head</span>(<span class="kw">read.table</span>(outfn, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>))
<span class="co">#&gt;                                filename col1 col2</span>
<span class="co">#&gt; 1 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    6    1</span>
<span class="co">#&gt; 2 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    7    2</span>
<span class="co">#&gt; 3 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    8    3</span>
<span class="co">#&gt; 4 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd    9    4</span>
<span class="co">#&gt; 5 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd   10    5</span>
<span class="co">#&gt; 6 /tmp/RtmpUrg5Db/file18ba912b1fa1_abcd   11    6</span></code></pre>
</div>
<div id="data-filtration" class="section level3">
<h3>Data filtration</h3>
<p>Data filtration or subset is an important step to clean or run the specific analysis. A series of data filtration function will be establish and fixed that can be re-used in the future.</p>
<pre class="sourceCode r"><code class="sourceCode r">demo_file &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">&quot;extdata&quot;</span>, <span class="st">&quot;demo/proteinpaint/fusions2pp_fusioncatcher.txt&quot;</span>, <span class="dt">package =</span> <span class="st">&quot;ngstk&quot;</span>)
input_data &lt;-<span class="st"> </span><span class="kw">read.table</span>(demo_file, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>, <span class="dt">header =</span> <span class="ot">TRUE</span>, <span class="dt">stringsAsFactors =</span> <span class="ot">FALSE</span>)
<span class="co"># Get data subset according the defined rule</span>
mhandler_extra_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_5 =</span> <span class="dv">1</span>, <span class="dt">gene_3 =</span> <span class="dv">2</span>, <span class="dt">any_gene =</span> <span class="st">&quot;TCF3&quot;</span>, <span class="dt">fusions_any_match_flag =</span> <span class="ot">TRUE</span>)
result_1 &lt;-<span class="st"> </span><span class="kw">fusions_filter</span>(input_data, <span class="dt">mhandler_extra_params =</span> mhandler_extra_params)
<span class="kw">head</span>(result_1)
<span class="co">#&gt;   gene5 gene3             fusion_type</span>
<span class="co">#&gt; 1  TCF3  PBX1                in-frame</span>
<span class="co">#&gt; 2  TCF3  PBX1                in-frame</span>
<span class="co">#&gt; 3  TCF3  PBX1                in-frame</span>
<span class="co">#&gt; 4  TCF3  PBX1 CDS(truncated)/intronic</span>
<span class="co">#&gt; 5  TCF3  PBX1 CDS(truncated)/intronic</span>

mhandler_extra_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_3 =</span> <span class="dv">2</span>, <span class="dt">right_gene =</span> <span class="st">&quot;GYPA&quot;</span>, <span class="dt">fusions_right_match_flag =</span> <span class="ot">TRUE</span>)
result_2 &lt;-<span class="st"> </span><span class="kw">fusions_filter</span>(input_data, <span class="dt">mhandler_extra_params =</span> mhandler_extra_params)
<span class="kw">head</span>(result_2)
<span class="co">#&gt;   gene5 gene3 fusion_type</span>
<span class="co">#&gt; 6  GYPE  GYPA    in-frame</span>
<span class="co">#&gt; 7  GYPE  GYPA    in-frame</span>

mhandler_extra_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_5 =</span> <span class="dv">1</span>, <span class="dt">left_gene =</span> <span class="st">&quot;GYPA&quot;</span>, <span class="dt">fusions_left_match_flag =</span> <span class="ot">TRUE</span>)
result_3 &lt;-<span class="st"> </span><span class="kw">fusions_filter</span>(input_data, <span class="dt">mhandler_extra_params =</span> mhandler_extra_params)
<span class="kw">head</span>(result_3)
<span class="co">#&gt; [1] gene5       gene3       fusion_type</span>
<span class="co">#&gt; &lt;0 rows&gt; (or 0-length row.names)</span>

mhandler_extra_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_5 =</span> <span class="dv">1</span>, <span class="dt">gene_3 =</span> <span class="dv">2</span>, <span class="dt">left_gene =</span> <span class="st">&quot;GYPE&quot;</span>, <span class="dt">right_gene =</span> <span class="st">&quot;GYPA&quot;</span>, <span class="dt">fusions_full_match_flag =</span> <span class="ot">TRUE</span>)
result_4 &lt;-<span class="st"> </span><span class="kw">fusions_filter</span>(input_data, <span class="dt">mhandler_extra_params =</span> mhandler_extra_params)
<span class="kw">head</span>(result_4)
<span class="co">#&gt;   gene5 gene3 fusion_type</span>
<span class="co">#&gt; 6  GYPE  GYPA    in-frame</span>
<span class="co">#&gt; 7  GYPE  GYPA    in-frame</span>

mhandler_extra_params =<span class="st"> </span><span class="kw">list</span>(<span class="dt">gene_5 =</span> <span class="dv">1</span>, <span class="dt">gene_3 =</span> <span class="dv">2</span>, <span class="dt">left_gene =</span> <span class="st">&quot;GYPE&quot;</span>, <span class="dt">right_gene =</span> <span class="st">&quot;GYPA&quot;</span>, <span class="dt">fusions_anyfull_match_flag =</span> <span class="ot">TRUE</span>)
result_5 &lt;-<span class="st"> </span><span class="kw">fusions_filter</span>(input_data, <span class="dt">mhandler_extra_params =</span> mhandler_extra_params)
<span class="kw">head</span>(result_5)
<span class="co">#&gt;   gene5 gene3 fusion_type</span>
<span class="co">#&gt; 6  GYPE  GYPA    in-frame</span>
<span class="co">#&gt; 7  GYPE  GYPA    in-frame</span></code></pre>
</div>
<div id="log-related" class="section level3">
<h3>Log related</h3>
<p>In the data analysis process, we usually need to check the create time and modification time of some result files, e.g.Â all of files create or change time are identical. ngstk provied <code>get_files_mtime</code> and <code>get_files_ctime</code> to process the files time check step. The default time check function is check all files time wheather are identical and you can use your own function.</p>
<pre class="sourceCode r"><code class="sourceCode r">file_a &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
file_b &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
<span class="kw">file.create</span>(<span class="kw">c</span>(file_a, file_b))
<span class="co">#&gt; [1] TRUE TRUE</span>
x &lt;-<span class="st"> </span><span class="kw">get_files_mtime</span>(<span class="dt">input_files =</span> <span class="kw">c</span>(file_a, file_b))
x
<span class="co">#&gt; $files_mtime</span>
<span class="co">#&gt; [1] &quot;2018-01-04 15:32:52 CST&quot; &quot;2018-01-04 15:32:52 CST&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $status</span>
<span class="co">#&gt; [1] TRUE</span>
x &lt;-<span class="st"> </span><span class="kw">get_files_mtime</span>(<span class="dt">input_files =</span> <span class="kw">c</span>(file_a, file_b), <span class="dt">return_check =</span> <span class="ot">FALSE</span>)
x
<span class="co">#&gt; [1] &quot;2018-01-04 15:32:52 CST&quot; &quot;2018-01-04 15:32:52 CST&quot;</span>
x &lt;-<span class="st"> </span><span class="kw">get_files_mtime</span>(<span class="dt">input_files =</span> <span class="kw">c</span>(file_a, file_b), <span class="dt">return_mtime =</span> <span class="ot">FALSE</span>)
x
<span class="co">#&gt; [1] TRUE</span>
x &lt;-<span class="st"> </span><span class="kw">get_files_ctime</span>(<span class="dt">input_files =</span> <span class="kw">c</span>(file_a, file_b))
x
<span class="co">#&gt; $files_ctime</span>
<span class="co">#&gt; [1] &quot;2018-01-04 15:32:52 CST&quot; &quot;2018-01-04 15:32:52 CST&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $status</span>
<span class="co">#&gt; [1] TRUE</span>
x &lt;-<span class="st"> </span><span class="kw">get_files_ctime</span>(<span class="dt">input_files =</span> <span class="kw">c</span>(file_a, file_b), <span class="dt">return_check =</span> <span class="ot">FALSE</span>)
x
<span class="co">#&gt; [1] &quot;2018-01-04 15:32:52 CST&quot; &quot;2018-01-04 15:32:52 CST&quot;</span></code></pre>
</div>
<div id="split-data" class="section level3">
<h3>Split data</h3>
<p>Split data is an optional step if you want to parallel process the data stream. ngstk provide <code>split_row_data</code> and <code>split_col_data</code> to split data.frame and data.table object.</p>
<pre class="sourceCode r"><code class="sourceCode r">x1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">col1 =</span> <span class="dv">1</span>:<span class="dv">39</span>, <span class="dt">col2 =</span> <span class="dv">1</span>:<span class="dv">39</span>)
x1
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 1     1    1</span>
<span class="co">#&gt; 2     2    2</span>
<span class="co">#&gt; 3     3    3</span>
<span class="co">#&gt; 4     4    4</span>
<span class="co">#&gt; 5     5    5</span>
<span class="co">#&gt; 6     6    6</span>
<span class="co">#&gt; 7     7    7</span>
<span class="co">#&gt; 8     8    8</span>
<span class="co">#&gt; 9     9    9</span>
<span class="co">#&gt; 10   10   10</span>
<span class="co">#&gt; 11   11   11</span>
<span class="co">#&gt; 12   12   12</span>
<span class="co">#&gt; 13   13   13</span>
<span class="co">#&gt; 14   14   14</span>
<span class="co">#&gt; 15   15   15</span>
<span class="co">#&gt; 16   16   16</span>
<span class="co">#&gt; 17   17   17</span>
<span class="co">#&gt; 18   18   18</span>
<span class="co">#&gt; 19   19   19</span>
<span class="co">#&gt; 20   20   20</span>
<span class="co">#&gt; 21   21   21</span>
<span class="co">#&gt; 22   22   22</span>
<span class="co">#&gt; 23   23   23</span>
<span class="co">#&gt; 24   24   24</span>
<span class="co">#&gt; 25   25   25</span>
<span class="co">#&gt; 26   26   26</span>
<span class="co">#&gt; 27   27   27</span>
<span class="co">#&gt; 28   28   28</span>
<span class="co">#&gt; 29   29   29</span>
<span class="co">#&gt; 30   30   30</span>
<span class="co">#&gt; 31   31   31</span>
<span class="co">#&gt; 32   32   32</span>
<span class="co">#&gt; 33   33   33</span>
<span class="co">#&gt; 34   34   34</span>
<span class="co">#&gt; 35   35   35</span>
<span class="co">#&gt; 36   36   36</span>
<span class="co">#&gt; 37   37   37</span>
<span class="co">#&gt; 38   38   38</span>
<span class="co">#&gt; 39   39   39</span>
x &lt;-<span class="st"> </span><span class="kw">split_row_data</span>(x1, <span class="dt">sections =</span> <span class="dv">2</span>)
x
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 1     1    1</span>
<span class="co">#&gt; 2     2    2</span>
<span class="co">#&gt; 3     3    3</span>
<span class="co">#&gt; 4     4    4</span>
<span class="co">#&gt; 5     5    5</span>
<span class="co">#&gt; 6     6    6</span>
<span class="co">#&gt; 7     7    7</span>
<span class="co">#&gt; 8     8    8</span>
<span class="co">#&gt; 9     9    9</span>
<span class="co">#&gt; 10   10   10</span>
<span class="co">#&gt; 11   11   11</span>
<span class="co">#&gt; 12   12   12</span>
<span class="co">#&gt; 13   13   13</span>
<span class="co">#&gt; 14   14   14</span>
<span class="co">#&gt; 15   15   15</span>
<span class="co">#&gt; 16   16   16</span>
<span class="co">#&gt; 17   17   17</span>
<span class="co">#&gt; 18   18   18</span>
<span class="co">#&gt; 19   19   19</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 20   20   20</span>
<span class="co">#&gt; 21   21   21</span>
<span class="co">#&gt; 22   22   22</span>
<span class="co">#&gt; 23   23   23</span>
<span class="co">#&gt; 24   24   24</span>
<span class="co">#&gt; 25   25   25</span>
<span class="co">#&gt; 26   26   26</span>
<span class="co">#&gt; 27   27   27</span>
<span class="co">#&gt; 28   28   28</span>
<span class="co">#&gt; 29   29   29</span>
<span class="co">#&gt; 30   30   30</span>
<span class="co">#&gt; 31   31   31</span>
<span class="co">#&gt; 32   32   32</span>
<span class="co">#&gt; 33   33   33</span>
<span class="co">#&gt; 34   34   34</span>
<span class="co">#&gt; 35   35   35</span>
<span class="co">#&gt; 36   36   36</span>
<span class="co">#&gt; 37   37   37</span>
<span class="co">#&gt; 38   38   38</span>
<span class="co">#&gt; 39   39   39</span>
x &lt;-<span class="st"> </span><span class="kw">split_row_data</span>(x1, <span class="dt">sections =</span> <span class="dv">3</span>)
x
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 1     1    1</span>
<span class="co">#&gt; 2     2    2</span>
<span class="co">#&gt; 3     3    3</span>
<span class="co">#&gt; 4     4    4</span>
<span class="co">#&gt; 5     5    5</span>
<span class="co">#&gt; 6     6    6</span>
<span class="co">#&gt; 7     7    7</span>
<span class="co">#&gt; 8     8    8</span>
<span class="co">#&gt; 9     9    9</span>
<span class="co">#&gt; 10   10   10</span>
<span class="co">#&gt; 11   11   11</span>
<span class="co">#&gt; 12   12   12</span>
<span class="co">#&gt; 13   13   13</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 14   14   14</span>
<span class="co">#&gt; 15   15   15</span>
<span class="co">#&gt; 16   16   16</span>
<span class="co">#&gt; 17   17   17</span>
<span class="co">#&gt; 18   18   18</span>
<span class="co">#&gt; 19   19   19</span>
<span class="co">#&gt; 20   20   20</span>
<span class="co">#&gt; 21   21   21</span>
<span class="co">#&gt; 22   22   22</span>
<span class="co">#&gt; 23   23   23</span>
<span class="co">#&gt; 24   24   24</span>
<span class="co">#&gt; 25   25   25</span>
<span class="co">#&gt; 26   26   26</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[3]]</span>
<span class="co">#&gt;    col1 col2</span>
<span class="co">#&gt; 27   27   27</span>
<span class="co">#&gt; 28   28   28</span>
<span class="co">#&gt; 29   29   29</span>
<span class="co">#&gt; 30   30   30</span>
<span class="co">#&gt; 31   31   31</span>
<span class="co">#&gt; 32   32   32</span>
<span class="co">#&gt; 33   33   33</span>
<span class="co">#&gt; 34   34   34</span>
<span class="co">#&gt; 35   35   35</span>
<span class="co">#&gt; 36   36   36</span>
<span class="co">#&gt; 37   37   37</span>
<span class="co">#&gt; 38   38   38</span>
<span class="co">#&gt; 39   39   39</span>
x1 &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">col1 =</span> <span class="dv">1</span>:<span class="dv">10</span>, <span class="dt">col2 =</span> <span class="dv">11</span>:<span class="dv">20</span>)
x1.t &lt;-<span class="st"> </span><span class="kw">t</span>(x1)
x &lt;-<span class="st"> </span><span class="kw">split_col_data</span>(x1.t, <span class="dt">sections =</span> <span class="dv">3</span>)
x
<span class="co">#&gt; [[1]]</span>
<span class="co">#&gt;      [,1] [,2] [,3]</span>
<span class="co">#&gt; col1    1    2    3</span>
<span class="co">#&gt; col2   11   12   13</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[2]]</span>
<span class="co">#&gt;      [,1] [,2] [,3]</span>
<span class="co">#&gt; col1    4    5    6</span>
<span class="co">#&gt; col2   14   15   16</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; [[3]]</span>
<span class="co">#&gt;      [,1] [,2] [,3] [,4]</span>
<span class="co">#&gt; col1    7    8    9   10</span>
<span class="co">#&gt; col2   17   18   19   20</span>
<span class="co"># split file</span>
dat &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">col1 =</span> <span class="dv">1</span>:<span class="dv">10000</span>)
outfn &lt;-<span class="st"> </span><span class="kw">tempfile</span>()
<span class="kw">write.table</span>(dat, outfn, <span class="dt">sep =</span> <span class="st">&quot;</span><span class="ch">\t</span><span class="st">&quot;</span>)
<span class="kw">split_row_file</span>(outfn)
<span class="co">#&gt; $`1`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_1&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`2`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_2&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`3`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_3&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`4`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_4&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`5`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_5&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`6`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_6&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`7`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_7&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`8`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_8&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`9`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_9&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`10`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_10&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`11`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_11&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`12`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_12&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`13`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_13&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`14`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_14&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`15`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_15&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`16`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_16&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`17`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_17&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`18`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_18&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`19`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_19&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`20`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_20&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`21`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_21&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`22`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_22&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`23`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_23&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`24`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_24&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`25`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_25&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`26`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_26&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`27`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_27&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`28`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_28&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`29`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_29&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`30`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_30&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`31`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_31&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`32`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_32&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`33`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_33&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`34`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_34&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`35`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_35&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`36`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_36&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`37`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_37&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`38`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_38&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`39`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_39&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`40`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_40&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`41`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_41&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`42`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_42&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`43`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_43&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`44`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_44&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`45`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_45&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`46`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_46&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`47`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_47&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`48`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_48&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`49`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_49&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`50`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_50&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`51`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_51&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`52`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_52&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`53`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_53&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`54`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_54&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`55`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_55&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`56`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_56&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`57`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_57&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`58`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_58&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`59`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_59&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`60`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_60&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`61`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_61&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`62`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_62&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`63`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_63&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`64`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_64&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`65`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_65&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`66`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_66&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`67`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_67&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`68`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_68&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`69`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_69&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`70`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_70&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`71`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_71&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`72`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_72&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`73`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_73&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`74`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_74&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`75`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_75&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`76`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_76&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`77`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_77&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`78`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_78&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`79`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_79&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`80`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_80&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`81`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_81&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`82`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_82&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`83`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_83&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`84`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_84&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`85`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_85&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`86`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_86&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`87`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_87&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`88`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_88&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`89`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_89&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`90`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_90&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`91`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_91&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`92`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_92&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`93`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_93&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`94`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_94&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`95`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_95&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`96`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_96&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`97`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_97&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`98`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_98&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`99`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_99&quot;</span>
<span class="co">#&gt; </span>
<span class="co">#&gt; $`100`</span>
<span class="co">#&gt; [1] &quot;/tmp/RtmpUrg5Db/file18ba964c4e8a6_split_100&quot;</span></code></pre>
</div>
<div id="filename-process" class="section level3">
<h3>Filename Process</h3>
<pre class="sourceCode r"><code class="sourceCode r">files_dir &lt;-<span class="st"> </span><span class="kw">system.file</span>(<span class="st">'extdata'</span>, <span class="st">'demo/format'</span>, <span class="dt">package =</span> <span class="st">'ngstk'</span>)
pattern &lt;-<span class="st"> '*.txt'</span>
<span class="kw">list.files</span>(files_dir, pattern)
<span class="co">#&gt; [1] &quot;-cancer-circrna.txt&quot;         &quot;common-circrna.txt&quot;         </span>
<span class="co">#&gt; [3] &quot;hg38_cancer_circrna_mre.txt&quot; &quot;hg38-cancer-circrna.txt&quot;    </span>
<span class="co">#&gt; [5] &quot;hg38-common-circrna.txt&quot;</span>
x &lt;-<span class="st"> </span><span class="kw">format_filenames</span>(<span class="dt">files_dir =</span> files_dir, <span class="dt">pattern =</span> pattern, <span class="dt">profix =</span> <span class="st">'hg38_'</span>)
x
<span class="co">#&gt; [1] &quot;/tmp/Rtmpd5dOjG/Rbuild18b98691c488a/ngstk/vignettes/hg38_cancer_circrna.txt&quot;    </span>
<span class="co">#&gt; [2] &quot;/tmp/Rtmpd5dOjG/Rbuild18b98691c488a/ngstk/vignettes/hg38_common_circrna.txt&quot;    </span>
<span class="co">#&gt; [3] &quot;/tmp/Rtmpd5dOjG/Rbuild18b98691c488a/ngstk/vignettes/hg38_cancer_circrna_mre.txt&quot;</span>
<span class="co">#&gt; [4] &quot;/tmp/Rtmpd5dOjG/Rbuild18b98691c488a/ngstk/vignettes/hg38_cancer_circrna.txt&quot;    </span>
<span class="co">#&gt; [5] &quot;/tmp/Rtmpd5dOjG/Rbuild18b98691c488a/ngstk/vignettes/hg38_common_circrna.txt&quot;</span></code></pre>
</div>
<div id="colors" class="section level3">
<h3>Colors</h3>
<pre class="sourceCode r"><code class="sourceCode r"><span class="kw">set_colors</span>(<span class="st">'default'</span>)
<span class="co">#&gt;  [1] &quot;#0073c3&quot; &quot;#efc000&quot; &quot;#696969&quot; &quot;#ce534c&quot; &quot;#7ba6db&quot; &quot;#035892&quot; &quot;#052135&quot;</span>
<span class="co">#&gt;  [8] &quot;#666633&quot; &quot;#660000&quot; &quot;#990000&quot;</span>
<span class="kw">set_colors</span>(<span class="st">'proteinpaint_mutations'</span>)
<span class="co">#&gt;  [1] &quot;#3987cc&quot; &quot;#ff7f0e&quot; &quot;#db3d3d&quot; &quot;#6633ff&quot; &quot;#bbbbbb&quot; &quot;#9467bd&quot; &quot;#998199&quot;</span>
<span class="co">#&gt;  [8] &quot;#8c564b&quot; &quot;#819981&quot; &quot;#5781ff&quot;</span>
<span class="kw">set_colors</span>(<span class="st">'proteinpaint_chromHMM_state'</span>)
<span class="co">#&gt;  [1] &quot;#c0222c&quot; &quot;#f12424&quot; &quot;#ff00c7&quot; &quot;#d192fb&quot; &quot;#f9982f&quot; &quot;#fcc88e&quot; &quot;#fbf876&quot;</span>
<span class="co">#&gt;  [8] &quot;#a6d67b&quot; &quot;#1fb855&quot; &quot;#007d37&quot; &quot;#00a99e&quot; &quot;#11aaec&quot; &quot;#186db9&quot; &quot;#3800f8&quot;</span>
<span class="co">#&gt; [15] &quot;#961a8b&quot; &quot;#47005f&quot;</span></code></pre>
</div>
</div>



<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
